File: CS/WindowsApplication3/Form1.cs
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;


namespace WindowsApplication3 {
    public partial class Form1: Form {
        public Form1() {
            InitializeComponent();
        }
        
        private void Form1_Load(object sender, EventArgs e) {
            DataHelper helper = new DataHelper(imageList1);
            DataSet dataSet1 = new DataSet();
            dataSet1.Tables.Add(helper.CreateTable());
            dataSet1.Tables.Add(helper.CreateEmptyTable());
            DataColumn keyColumn = dataSet1.Tables[0].Columns["CustomerID"];
            DataColumn foreignKeyColumn = dataSet1.Tables[1].Columns["CustomerID"];
            dataSet1.Relations.Add("CustomerInfo_CustomerInfo1", keyColumn, foreignKeyColumn);
            gridControl1.DataSource = dataSet1.Tables[0];

            RepositoryItemCustomEdit edit = new RepositoryItemCustomEdit();
            gridControl1.RepositoryItems.Add(edit);
            colFirstName.ColumnEdit = edit;
           
        }

        
    }
}

File: VB/WindowsApplication3/Form1.vb
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms


Namespace WindowsApplication3
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			Dim helper As New DataHelper(imageList1)
			Dim dataSet1 As New DataSet()
			dataSet1.Tables.Add(helper.CreateTable())
			dataSet1.Tables.Add(helper.CreateEmptyTable())
			Dim keyColumn As DataColumn = dataSet1.Tables(0).Columns("CustomerID")
			Dim foreignKeyColumn As DataColumn = dataSet1.Tables(1).Columns("CustomerID")
			dataSet1.Relations.Add("CustomerInfo_CustomerInfo1", keyColumn, foreignKeyColumn)
			gridControl1.DataSource = dataSet1.Tables(0)

			Dim edit As New RepositoryItemCustomEdit()
			gridControl1.RepositoryItems.Add(edit)
			colFirstName.ColumnEdit = edit

		End Sub


	End Class
End Namespace
